Era customizations for the GEM-CSC and CSC-RPC local integrated triggers for Phase2

dildick opened the pull request:
Previous attempt (https://github.com/cms-sw/cmssw/pull/13364) had changes in both Run2 and Phase2 scenarios. This PR enables the local integrated triggers only for the Phase2 and Phase2_dev scenarios. It should not interfere with Run2_25ns. To be integrated after https://github.com/cms-sw/cmssw/pull/13963.

Remaining issues in https://github.com/cms-sw/cmssw/pull/13364 (see below) will be addressed in another PR.
- isSLHC should be False for Run2_25ns
- addition of a "muonTrigger2017" scenario for high PU running (with isSLHC set to True), i.e. Run2_2017, Run2_2017_trackingPhase1PU70 and Run2_2017_trackingRun2

davidlange6 committed:
resolve conflict

Commit:
Refactor BXVector::set() to replace erase/insert.

Using erase()/insert() causes massive overhead, much faster with at()
This particularly affects the L1TCaloLayer1 module.

l-cadamuro committed:
make tau vars methods const

cferraio committed:
Added zdc unpacker to 81x and fixed digi->reco step

cferraio committed:
adding ZdcUnpacker.cc(h)

perrotta committed:
Update up to /dev/CMSSW_8_0_0/HLT/V87

cferraio committed:
fixed for MC generation

perrotta committed:
Update the HLT menu up to /dev/CMSSW_8_0_0/HLT/V88

davidlange6 committed:
move to 2016 era and 2016 hlt where it is working

davidlange6 committed:
fix typo

cferraio committed:
minor edits based on PR comments

Martin-Grunewald committed:
Rebase to CMSSW_8_1_X_2016-04-04-2300

Martin-Grunewald committed:
Subtables

davidlange6 committed:
more era migration

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V98

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V98

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V105

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V105

cferraio committed:
zdc hit reconstructor looks for hcal digis, then castor digis for reco

Martin-Grunewald committed:
Rebase to 804

Martin-Grunewald committed:
Rebase to 804

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V111

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V111

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V115

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V115

mulhearn committed:
CondFormats updates for l1t-tsg-v5

mulhearn committed:
unpacker updates for l1t-tsg-v5

mulhearn committed:
Instead of provenance, use a config setting for unpacking mask in L1TCaloLayer1

rekovic committed:
Add customisation to re-emulate L1 from MC RAW and use simCscTriggerPrimitiveDigis

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V119

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V119

Martin-Grunewald committed:
Fix fastsim for CMSHLT-729

Martin-Grunewald committed:
Fix fastsim for CMSHLT-729

lgray committed:
add HGCal to the geometries with Phase2 Tracker in them

ianna committed:
* Make sure upgrade workflow does not crash due to serious overlaps

perrotta committed:
Upate HLT menu up to /dev/CMSSW_8_0_0/HLT/V127

perrotta committed:
Upate HLT menu up to /dev/CMSSW_8_0_0/HLT/V127

cmsbuild commented:
A new Pull Request was created by  @dildick (Sven Dildick) for CMSSW_8_1_X.

It involves the following packages:

L1Trigger/CSCTriggerPrimitives


@cmsbuild, @rekovic, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@valuev, @Martin-Grunewald this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

perrotta committed:
Remove CMSHLT-716 from the subtables and update HLT menu up to /dev/CMSSW_8_0_0/HLT/V228

perrotta committed:
Remove CMSHLT-716 from the subtables and update HLT menu up to /dev/CMSSW_8_0_0/HLT/V228

Commit:
Fix compilation with Clang (3.7.1)

The patch resolves issue with Clang compiler.

N3690 (should be C++11 standard) and latest draft N4567, 9.4.2/3

...
The member shall still be defined in a namespace scope if it
is odr-used (3.2) in the program and the namespace scope definition
shall not contain an initializer.
9.4.2/4 talks about how const static data members are being handled.

Also standard says that no diagnostic is required by compiler.

The following undefined references were reported by linker:

    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x61): undefined reference to GCTriggerDetId::cell_mask'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x6e): undefined reference to GCTriggerDetId::cell_shift'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x7b): undefined reference to GCTriggerDetId::sector_mask'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x85): undefined reference to GCTriggerDetId::sector_shift'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x94): undefined reference to GCTriggerDetId::module_mask'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0x9e): undefined reference to GCTriggerDetId::module_shift'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0xad): undefined reference to GCTriggerDetId::layer_mask'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0xb7): undefined reference to GCTriggerDetId::layer_shift'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0xc6): undefined reference to GCTriggerDetId::zside_mask'
    DataFormats/ForwardDetId/src/HGCTriggerDetId.cc:(.text+0xcf): undefined reference to GCTriggerDetId::zside_shift'

These are odr-used because

    inline void setMaskedId( const uint32_t value, const uint32_t &shift, const uint32_t &mask )

2nd and 3rd arguments are passed by a reference.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

VinInn committed:
remove memory fence

avetisya committed:
Reinitializing deleted pointers to nullptr

hengne committed:
update gensim input for 2017 workflows

Martin-Grunewald committed:
Fixup crash observed in 0-event 1307.0 due to L1TGlobalSummary/L1TGlobalUtil

Martin-Grunewald committed:
Fixup crash observed in 0-event 1307.0 due to L1TGlobalSummary/L1TGlobalUtil

Martin-Grunewald committed:
Fixup crash observed in 0-event 1307.0 due to L1TGlobalSummary/L1TGlobalUtil

hengne committed:
fix typo in gensim input string

hengne committed:
change to phase1 design GT with new gensim input

mmusich committed:
resolve conflict

jdolen committed:
Fixing filenames to match cms-data

jdolen committed:
Fixing filenames to match cms-data

jan-kaspar committed:
Merge branch 'CMSSW_8_1_X' of github.com:CTPPS/cmssw into ctpps_raw_cond_formats
Need merged DetId and raw-data data formats.

jan-kaspar committed:
Merge branch 'CMSSW_8_1_X' of github.com:CTPPS/cmssw into ctpps_raw_cond_formats

Conflicts:
	CondFormats/TotemReadoutObjects/BuildFile.xml
	CondFormats/TotemReadoutObjects/src/TotemFramePosition.cc
	DataFormats/L1THGCal/BuildFile.xml
	DataFormats/TotemDigi/BuildFile.xml
	DataFormats/TotemDigi/interface/TotemVFATStatus.h
	DataFormats/TotemDigi/src/TotemVFATStatus.cc
	DataFormats/TotemRPDetId/BuildFile.xml
	DataFormats/TotemRPDetId/src/classes.h
	DataFormats/TotemRPDetId/src/classes_def.xml
	DataFormats/TotemRawData/BuildFile.xml

ndaci committed:
Remove harmful consumes call inside of a loop.

Dr15Jones committed:
Streamline calling of unscheduled EDProducers

When a data product from an unscheduled EDProducer was requested, we used to have UnscheduledProductResolver call a virtual function of Principal to call a virtual function of UnscheduledHandler to trigger looking the Worker which then calls the module. Now we setup to have the UnscheduledProductResolver have direct access to the Worker and therefore invoke it directly without any lookup or extra virtual calls.
This refactoring is needed for work to allow asynchronous calling of unscheduled EDProducers.

davidlange6 committed:
Merge pull request #14070 from VinInn/Ref81

remove memory fence

cmsbuild committed:
Merge pull request #14069 from davidlt/fix-HGCTriggerDetId-clang

Fix compilation with Clang (3.7.1)

cmsbuild committed:
Merge pull request #14067 from avetisya/nullptr-reinitialization-fix

Reinitializing deleted pointers to nullptr

cmsbuild committed:
Merge pull request #14054 from nsmith-/layer1Unpacker-fix

L1TCalo Layer 1 guard against no FED data in event

davidlange6 committed:
Merge pull request #13928 from davidlt/gcc600-reco-maybe-uninitialized

Default initialize PTrajectoryStateOnDet members

davidlange6 committed:
Merge pull request #14055 from smuzaffar/runtestTqafTopEventSelection-unittest-fix

Fix for runtestTqafTopEventSelection unit test

boudoul committed:
moved to External

davidlange6 committed:
Merge pull request #13923 from boudoul/workflowsp2_81X_localreco

2023LReco and 2023dev scenarios defined up to local reco

Commit:
Add GE1/1 local trigger in foreseen Run3 scenario.

mtosi committed:
follow suggestion from carl (thanks) and re-write an 'if statement'

davidlange6 committed:
Merge pull request #14082 from slava77/cferraio-ZdcUnpacker_DigiRecoFix81X

Zdc unpacker digi reco (rebase of #13882)

davidlange6 committed:
Merge pull request #14014 from makortel/inlineLiblongintpack

Declare liblogintpack functions as inline

mmusich committed:
Adding payloads for tracking MVA at HLT

Commit:
add templated filldescription functions for T1MET at HLT

rekovic committed:
Revert Layer-1 corrections for high-eta, for consistency with current default calibrations

Dr15Jones committed:
Filled needed for previous commit

These files are needed for the simplification of unscheduled calling.

mmusich committed:
resolve conflict

davidlange6 committed:
logic bug fixes identified by valgrind

davidlange6 committed:
logic bug fixes identified by valgrind

mmusich committed:
revert bad update

davidlange6 committed:
Merged refs/pull/13862/head from repository cms-sw

davidlange6 committed:
Merged refs/pull/13893/head from repository cms-sw

hengne committed:
Merge branch 'from-CMSSW_8_1_0_pre2_relval', remote branch 'mengqing/from-CMSSW_8_1_0_pre2_relval' into from-CMSSW_8_1_0_pre2_relval

hengne committed:
clean up

hengne committed:
fix conflict

hengne committed:
clearn up

hengne committed:
Merge branch 'from-CMSSW_8_1_0_pre2_relval_DigiSigTest' into from-CMSSW_8_1_0_pre2_relval_test

fwyzard committed:
hltDumpStream: migrate to L1 Stage 2

hengne committed:
update data 2015d and fastsim workflows to use relval2016 HLT menu

davidlange6 committed:
Merge pull request #13962 from jhgoh/PseudoTopFix76

Synchronize pseudo top definition with LHCTOPWG - 81X

cmsbuild committed:
Merge pull request #13637 from mtosi/from-CMSSW_8_1_X_2016-03-06-2300_stoppingSource

add monitoring of tracks stopping source

mulhearn committed:
bring L1T algorithm up to date with l1t-integration-v34.0

hengne committed:
resolve confilict

davidlange6 committed:
Merged refs/pull/13940/head from repository cms-sw

Martin-Grunewald committed:
Sync master and subtables up to CMSHLT-754

Martin-Grunewald committed:
Sync master and subtables up to CMSHLT-754

davidlange6 committed:
Merge pull request #13940 from mdhildreth/PreMixing_Stage2L1Fix80X

Add eras switch for L1 configuration for PreMixing

hengne committed:
resolve conflict

hengne committed:
resolve conflict

hengne committed:
update phase1 gensim string

hengne committed:
clean up

kirschen committed:
restrict NHF-cut to eta<2.4 to prevent it from being applied in the HF

kirschen committed:
restrict NHF-cut to eta<2.4 to prevent it from being applied in the HF

Martin-Grunewald committed:
Migration to L1T v5 menu

Martin-Grunewald committed:
Migration to L1T v5 menu

cmsbuild committed:
Merge pull request #14086 from davidlange6/l1fixes81x

L1T logic bug fixes

cmsbuild committed:
Merge pull request #14087 from fwyzard/hltDumpStream_migrate_to_L1Stage2_80x

hltDumpStream: migrate to L1 Stage 2

hengne committed:
change L1REPACK:GT2 to L1REPACK:Full for stage-2 L1

hengne committed:
era change to Run2_2016

hengne committed:
update era to Run2_2016

alja committed:
Don't open file dialog when geom browsing configuration is set.

hengne committed:
revert back, to the version wo PR14077

hengne committed:
add back T2_CH_CERN requirement

Martin-Grunewald committed:
cleanup of autoCondHLT

Martin-Grunewald committed:
cleanup of autoCondHLT

kpedro88 committed:
separate CaloHitResponse for QIE10

kpedro88 committed:
separate CaloHitResponse for QIE10

Martin-Grunewald committed:
Update autoCond.py to #14077

Commit:
Bug fix for RECO geometry of HCAL

Dr15Jones committed:
Fix multiple include barrier for UnscheduledAuxiliary.h

The define statement did not match the CPP ifndef statement. The problem was found by clang.

kpedro88 committed:
fix HcalDbHardcode pedestal bug, address a few lingering code review comments

davidlange6 committed:
Merge pull request #14053 from ianna/2023dev-scenario-fix

2023dev Scenario Fix

Martin-Grunewald committed:
GT update with #13893 plus further update as in #14077

cmsbuild committed:
Merge pull request #14081 from Dr15Jones/simplifyUnscheduledCalling

Streamline calling of unscheduled EDProducers

jdolen committed:
Fixing PUID working points 

Calculated using the correct input file (80X trained PUMVA)

jdolen committed:
Fixing PUID working points

Calculated using the correct input file (80X trained PUMVA)

hengne committed:
fix harvetst step name change in highstats workflows

Commit:
small fix

davidlange6 committed:
Merge pull request #14107 from davidlange6/l1hlt80x

HLT for L1v4 + GT+ relval + L1 fixes from valgrind (back ports)

davidlange6 committed:
Merge pull request #14096 from alja/geo-browse-open-file-dialog

80x Fireworks: open file dialogue & geometry browsing

davidlange6 committed:
Merge pull request #14071 from avetisya/json-mon-final-stage2-adaptation

Reinitializing deleted pointers to nullptr (80X)

mulhearn committed:
disable HF in Layer1 if HCAL geometry condition has use1x1 false

uhussain committed:
PackedCandidate closest vertex

uhussain committed:
drawModule vhbot removed from line 554

rekovic committed:
Sync egIsoLUTFile from caloStage2Params_2016_v2_cfi.py and caloStage2Params_cfi.py

Commit:
Merge remote branch 'origin/CMSSW_8_0_X' into CMSSW_8_1_X

Conflicts:
	Configuration/AlCa/python/autoCond.py

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

mulhearn committed:
Provide new CondFormat functionality of l1t-tsg-v5, without changing CondFormats

davidlange6 committed:
Merge pull request #14062 from ianna/ddvalue-trivial-destructor

Let the Compiler Provide the Default Implementation of the Destructor

davidlange6 committed:
Merge pull request #14049 from jdolen/puid_81x

Update PUID to 80X training (PR81X)

davidlange6 committed:
Merge pull request #14060 from rovere/checkUpdatedStateValidity

Protect access to updatedState

davidlange6 committed:
Merge pull request #14043 from jdolen/puid_80x_try2

Update PUID to 80X training

davidlange6 committed:
Merge pull request #14039 from cms-met/fillDescHLTMet8XY

add fillDescriptions function needed by the HLT

davidlange6 committed:
Merge pull request #13954 from cms-met/puppiMetFix803

Fix Puppi MET and smearing MET correction, add central test for the MET tool

davidlange6 committed:
Merge pull request #13906 from cms-l1t-offline/extcond-trig-l1t-80x

External conditions (technical) triggers for L1 Stage2.

davidlange6 committed:
Merge pull request #14033 from lgray/hgcal_digi_speed_and_rechitE

Use bin center for HGCal UncalibHits + Speedup in Digitizer

davidlange6 committed:
Merge pull request #14040 from cms-met/fillDescHLTMet8XY

add fillDescriptions function needed by the HLT (81X)

Commit:
Merge remote branch 'origin/CMSSW_8_0_X' into CMSSW_8_1_X

Conflicts:
	RecoJets/JetProducers/python/PileupJetIDCutParams_cfi.py
	RecoJets/JetProducers/python/PileupJetIDParams_cfi.py
	TrackingTools/PatternTools/src/TempTrajectory.cc
	TrackingTools/PatternTools/src/Trajectory.cc

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

alberto-sanchez committed:
adding phi meson

cmsbuild committed:
Merge pull request #14095 from kirschen/CMSSW_8_1_X_HLTPFJetId_FixNHF

restrict NHF-cut to eta<2.4 to prevent it from being applied in the HF

cmsbuild committed:
Merge pull request #14073 from Martin-Grunewald/FixL1TGlobal81X

Fixup crash observed in 0-event 1307.0 due to L1TGlobalSummary|Util

kpedro88 committed:
restore tilted tracker in 2023dev

lgray committed:
Merge pull request #22 from kpedro88/workflowsp2_81X_localreco

restore tilted tracker in 2023dev

Commit:
fix memory leak

Commit:
fix memory leak

Commit:
fix memory leak

lathomas committed:
Update following Slava's comments

davidlange6 committed:
Merge pull request #14061 from ndaci/PR_steam_fix_81X

Fix getByLabelWithoutRegistration (81X).

davidlange6 committed:
Merge pull request #14120 from lveldere/fastsim-80X-memleak

fix memory leak

smorovic committed:
Handling case when due to high ramdisk access latency (exceptional condition that does not happen normally, but is a result of some other problem), LS value given by CMSSW_STOP file
(created by hltd usually when cloud switch is made mid-run) is older than the current LS processed by CMSSW. In that case it was possible to end run in the mid of current LS, which
rarely resulted in not writing EoLS file to disk and thus causing lumisection to remain unmerged on FU. This patch sets higher "stop LS" if in such condition.

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

alja committed:
Fix redraw problems when change collection properties.

wmtan committed:
Use unique_ptr, not auto_ptr

cmsbuild committed:
Merge pull request #14104 from cms-l1t-offline/pr-l1t-integration-v34.0-81x

Bring L1T (nearly) too l1t-tsg-v5 (81x version)

cmsbuild committed:
Merge pull request #14101 from kpedro88/Phase1-HF7

HCAL Phase1 bug fixes

davidlange6 committed:
Merge pull request #14020 from bsunanda/Phase2-hgx48

Phase2-hgx48 Introduce fiducial cut off for HGC SimHit

davidlange6 committed:
Merge pull request #14074 from Martin-Grunewald/FixL1TGlobal80X

Fixup crash observed in 0-event 1307.0 due to L1TGlobalSummary|Util 80X

cmsbuild committed:
Merge pull request #14088 from fwyzard/hltDumpStream_migrate_to_L1Stage2_80x

hltDumpStream: migrate to L1 Stage 2

cmsbuild committed:
Merge pull request #14094 from kirschen/CMSSW_8_0_X_HLTPFJetId_FixNHF

restrict NHF-cut to eta<2.4 to prevent it from being applied in the HF

cmsbuild committed:
Merge pull request #14128 from alja/redraw

80x Fireworks: Fix redraw problems when change collection properties.

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

cmsbuild committed:
Merge pull request #14103 from kpedro88/Phase1-HF7-80X

separate CaloHitResponse for QIE10

davidlange6 committed:
Merge pull request #14125 from smorovic/fix-cloud-stop-80X

fix closing lumisection in HLT(DAQ) when stop requested for Cloud switchover (80X)

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

cmsbuild committed:
Merge pull request #14100 from bsunanda/Phase2-hgx49

bsunanda:Phase2-hgx49 Bug fix for RECO geometry of HCAL

cmsbuild committed:
Merge pull request #13828 from hqucms/SiStripO2O

New codes for Strip O2O

cmsbuild committed:
Merge pull request #14052 from nsmith-/layer1Unpacker-fix

L1TCalo Layer 1 unpacker guard against no FED data in event

davidlange6 committed:
Merge pull request #13838 from CTPPS/ctpps_raw_cond_formats

CTPPS: new CondFormats related to TOTEM RP raw data and digi

kpedro88 committed:
fix up zero suppression for QIE10

cmsbuild committed:
Merge pull request #14122 from lveldere/fastsim-81X-memleak

fix memory leak

Commit:
Remove NoHalo from NoBPTX path names

cmsbuild committed:
Merge pull request #13980 from mtosi/from-CMSSW_8_0_3_patch1_updateTrackCutSelector

add selection based on PV error (needed by HI @HLT)

davidlange6 committed:
Merge pull request #14130 from wmtan/UniquePtrInFastSim

Use unique_ptr, not auto_ptr in FastSimulation

lathomas committed:
Update

hengne committed:
remove frontier digi sig ib test workflows

hengne committed:
use era in data reco step

mmusich committed:
update Run1 and Run2 GT in autoCond for 50ns JEC uncertaities

davidlange6 committed:
Merge pull request #14152 from mmusich/80X_Fix50nsUncertainties

Update Run1 and Run2 GT in autoCond for 50ns JEC uncertaities

cmsbuild committed:
Merge pull request #14139 from kpedro88/Phase1-HF8

fix up zero suppression for QIE10

cmsbuild committed:
Merge pull request #14119 from lgray/workflowsp2_81X_localreco

Add HGCal to the geometries with Phase2 Tracker

cmsbuild committed:
Merge pull request #14099 from cms-tsg-storm/HLTinApril81X

HLT update to L1T menu v5 incl. GT

davidlange6 committed:
Merge pull request #13992 from lgray/hgcal_eras

HGCal Eras

hengne committed:
change fastSim HLT:@relval2016 back to relval25ns, because it meet a crash l1t::L1TGlobalUtil::retrieveL1Event

cmsbuild committed:
Merge pull request #13981 from mtosi/from-CMSSW_8_0_3_patch1_updateTrackCutSelector

add selection based on PV error (needed by HI @HLT)

davidlange6 committed:
Merge pull request #14111 from uhussain/CMSSW_8_1_X

Fixed static analyzer warning in PuppiProducer.cc & TrackerMap.cc

cmsbuild committed:
Merge pull request #13944 from barvic/csc_offline_dqm_added_timing_plots_81X

CSCOfflineMonitor DQM module - added timing plots 81X

davidlange6 committed:
Merge pull request #14132 from davidlange6/for803p2

Add RAWAODSIM event content definition

hengne committed:
add back HLT:@relval2016 for fastsim noPU and classical PU workflows, only fastsim pu premixing workflows still have problem

Commit:
Merge remote branch 'origin/CMSSW_8_0_X' into CMSSW_8_1_X

Conflicts:
	Configuration/AlCa/python/autoCond.py

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

hengne committed:
add hlt @relval2016 also for fastsim pu premixing

Dr15Jones committed:
Updated WaitingTaskList to handle exception

If the task waited upon throws an exception, that exception is communicated to the waiting tasks. This requires a new base class for tasks, WaitingTask.
The unit test was update to reflect the new API and to test the exception case.

hengne committed:
add DIGI2RAW before L1Reco for all fastSim workflows

wmtan committed:
Add missing std::move()

lathomas committed:
Update, following comments made on the 81X version of this PR (PR nb 13926)

Martin-Grunewald committed:
Revert GT update from #14152

davidlange6 committed:
Merge pull request #13926 from lathomas/CMSSW_8_1_X

Update of the Beam Halo Filter (both CSC and Calo algos)

davidlange6 committed:
Merge pull request #14023 from lathomas/CMSSW_8_0_X

Beam Halo Filter update 80X

mmusich committed:
Fixing snapshot times of Global Tag in #14152

davidlange6 committed:
Merge pull request #14156 from mmusich/80X_data_fixSnapshotTime

Fixing snapshot times of Global Tag in #14152

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

Martin-Grunewald committed:
Move GTs to #14156 versions

davidlange6 committed:
Merge pull request #14106 from cms-tsg-storm/HLTinApril80X

HLT update to L1T menu v5 incl. GT (80X)

cmsbuild committed:
Merge pull request #14137 from cms-l1t-offline/pr-l1t-tsg-v5-cand-81x

Bring L1T up-to-date with l1t-tsg-v5-cand

cmsbuild committed:
Merge pull request #14036 from jruizvar/addExoPaths81X

PFMET paths for Monojet EXO DQM + Remove NoHalo (81X)

Spudmeister committed:
uMNio data skipping implemented

kpedro88 committed:
enable full localreco and Phase2 era in upgrade workflows

kpedro88 committed:
additional zdc Phase2 custom

kpedro88 committed:
ECAL customs to fix exceptions in Phase2 workflows

kpedro88 committed:
rename 2023dev -> 2023tilted

kpedro88 committed:
fix from @jshlee for ME0 crash

Commit:
Bug fix for relabelling eta/phi index

mmusich committed:
Add HcalIsoTrkFilter to SingleElectron PD

mmusich committed:
Add HcalIsoTrkFilter to SingleElectron PD

mmusich committed:
Align Global Tags to 80X and introduce new GTs for HLT

mmusich committed:
Align Global Tags to 80X and introduce new GTs for HLT

davidlange6 committed:
Merge pull request #14155 from wmtan/AddMissingMove

Add missing std::move() in LuminosityBlock::put()

kpedro88 committed:
bunchSpacingProducer in localreco for phase2

kpedro88 committed:
bunchSpacingProducer in localreco for all

Dr15Jones committed:
Added missing header to WaitingTask.h

needed to add <memory> include in order to allow clang to compile.

Commit:
git rid of uninitialised value in FBaseSimEvent

Commit:
git rid of uninitialised value in FBaseSimEvent

davidlange6 committed:
Merge pull request #14173 from lveldere/fastsim-fbasesimevent-fix-81X

FastSim: bug fix: get rid of uninitialised value in FBaseSimEvent

cmsbuild committed:
Merge pull request #14158 from UMN-CMS/HcalUnpackerPatch

uMNio data skipping implemented for urgent HLT patch

cmsbuild committed:
Merge pull request #14153 from Dr15Jones/updateWaitingTaskList

Updated WaitingTaskList to handle exception

cmsbuild committed:
Merge pull request #14058 from VinInn/SpeedPFC81_2

speed up avoiding eta phi computations

cmsbuild committed:
Merge pull request #14048 from VinInn/SpeedECAL81

Speedup ECAL (for HLT)

davidlange6 committed:
Merge pull request #14098 from hengne/from-CMSSW_8_1_X_2016-04-14-2300_relval

relval matrix update + data relval HLT:@relval2016

mmusich committed:
Merge branch 'GTs_81X_andAutoAlca' of https://github.com/mmusich/cmssw into GTs_81X_andAutoAlca

Commit:
FastTrackerRecHit id is index in FastTrackerRecHitCollection

davidlange6 committed:
Merged refs/pull/14098/head from repository cms-sw

davidlange6 committed:
Merged refs/pull/14165/head from repository cms-sw

davidlange6 committed:
Merge pull request #14116 from alberto-sanchez/addphi

adding phi meson

davidlange6 committed:
add some workflows from lreco and tilted tracker

cmsbuild committed:
Merge pull request #14166 from mmusich/GTs_81X_andAutoAlca

Update HLT workflows as at P5 and fix autoAlca

davidlange6 committed:
Merge pull request #14171 from lveldere/fastsim-fbasesimevent-fix-80X

FastSim bugfix : get rid of uninitialised value in FBaseSimEvent

davidlange6 committed:
Merge pull request #14174 from lveldere/fastsim-fix-recHitId

FastSim bug fix: proper FastTrackerRecHit id

vanbesien committed:
Updating scenario to ppEra_Run2_2016

Martin-Grunewald committed:
Fix Unrunnable Schedule due to multiple offlineBeamSpots

cmsbuild committed:
Merge pull request #14164 from bsunanda/Run2-hcx77

bsunanda:Run2-hcx77 Bug fix for relabelling eta/phi index

davidlange6 committed:
Merge pull request #13563 from jruizvar/oldFiles81X

Remove obsolete files in the HLT SUSY Validation package (forward port of #13460)

Commit:
Merge remote branch 'origin/CMSSW_8_0_X' into CMSSW_8_1_X

Conflicts:
	Configuration/AlCa/python/autoCond.py
	FastSimulation/TrackingRecHitProducer/plugins/SiTrackerGaussianSmearingRecHitConverter.cc

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

davidlange6 committed:
Merge pull request #14181 from davidlange6/addPhIIscenarios

Add phase ii scenarios to runTheMatrix on top of #14098 #14165

cmsbuild committed:
Merge pull request #14192 from vanbesien/online_DQM_config_update_ppEra_Run2_2016

Updating scenario to ppEra_Run2_2016 - 81X

cmsbuild committed:
Merge pull request #14190 from Martin-Grunewald/UnRunOff81X

Fix Unrunnable Schedule due to multiple offlineBeamSpots (81X)

Commit:
Merged for-GEM-CSC-RPC-trigger-eras-customization from repository dildick

Commit:
proper use of eras

cmsbuild commented:
Pull request #14056 was updated. @covarell, @lveldere, @perrozzi, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @govoni, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @deguio, @slava77, @ggovi, @mmusich, @mulhearn, @ssekmen, @rekovic, @emeschi, @thuer, @dmitrijus, @alja, @srimanob, @hengne can you please check and sign again.

dildick commented:
A lot of work is needed before this can go in
